### ISSUE TITLE
the method of counting columns countSourceCols() is not clearly described

### ISSUE THREAD

**Issue by aninde** [handsontable/docs#120]
We should complete the description regarding the `data` and `countSourceCols()`

Without adding `content: null` to the `data`, `countSourceCols` omits empty cells, which can be very confusing to our users.

We should add for countSourceCols()info about that it will take value either from `schema`, `columns`or the FIRST row from the data set.

Explanation and discussion is here
https://github.com/handsontable/handsontable/issues/4176#issuecomment-653493968

**Comment by scarletfog** [comment]
I looked through the comments in the hot repo, looks like 8.0.0 fixes something here - is it still an issue @aninde ?

**Comment by aninde** [comment]
Yes it is still valid, v8 did not make any changes here. As it was written in the comments, this report is a way to resolve Case 1 reported https://github.com/handsontable/handsontable/issues/4176#issuecomment-653493968. This requires the documentation to be completed as it is written here https://github.com/handsontable/docs/issues/120#issue-650569106.

**Comment by scarletfog** [comment]
Ok, thanks @aninde ! I looked at the issue and Ola wrote about the case being fixed by 8.0.0 so I thought the behavior is more predictable now. I will dig deeper into the topic.
